lib/kafka/producer_manager.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require 'singleton'
+require 'waterdrop'
+
+module Kafka
+  class ProducerManager
+    include Singleton
+
+    attr_reader :producer
+
+    def initialize
+      setup_producer if Flipper.enabled?(:kafka_producer)
+    end
+
+    private
+
+    def setup_producer
+      @producer = WaterDrop::Producer.new do |config|
+        config.deliver = true
+        config.kafka = {
+          'bootstrap.servers': Settings.kafka_producer.broker_urls.join(','),
+          'request.required.acks': 1,
+          'message.timeout.ms': 100
+        }
+        config.logger = Rails.logger
+        config.client_class = if Rails.env.test?
```

Comment:
Overall looking good. I just see a couple more spec opportunities, could we add a spec to test the different clients behavior that changes on different environments? could use `allow(Rails.env).to receive(:test?).and_return(false)`

Reply:
Yes, good idea. I'll try and add a spec about the client class depending on the environment. I'll need to set it up so as to prevent it from actually trying to connect to a cluster, though.

Reply:
I added some specs for this but it turned out trickier than expected. The reason was mainly the fact that we use the Singleton pattern for the `Kafka::ProducerManager`. I'll create a followup ticket to experiment with other patterns for ensuring that we have a single, long-lived instance of the underlying `WaterDrop::Producer`.